### DISCUSSION TITLE
Hvordan holder vi hverandre oppdatert på nye komponenter?

### DISCUSSION THREAD

**Discussion by onordbo** [fremtind/jokul#2362] [Annet]
Stadig hører jeg fra både designere og utviklere at nye komponenter som lanseres både i Figma og biblioteket, går oss hus forbi. Trenger vi ett lettbeint fora _(alá en slags changelog)_ for å kunne dele hva som er nytt? Hvordan burde dette se ut, og hvor skal den leve? Jeg tror ett viktig kriterie for at vi skal god verdi av en slik funksjon er at den må være lett å oppdatere, så alle kan gjøre det.

Kjør debatt!

**Comment by AudunBerdal** [comment]
Jeg synes vi i større grad burde legge til en beskrivelse når vi publiserer endringer til komponentbiblioteket i Figma. Da får alle designere en beskrivelse av hva som har endret seg før man oppdaterer. Dette vil også legge seg i versjonshistorikken, selvom denne riktignok ligger litt skjult. Kanskje finnes det en mulighet for å kunne hente ut versjonshistorikken fra kompontentbibliotek-fila i Figma med noe kodemagi og presentere den et mer tilgjengelig sted?

<img width="225" alt="Skjermbilde 2021-10-08 kl  12 17 04" src="https://user-images.githubusercontent.com/57794116/136539805-201863bb-a850-4f49-baa2-e5d53a1b681d.png">
?

**Reply by wkillerud** [reply]
[Pluralsight](https://www.figma.com/community/file/915654353202115828) og [Moon](https://www.figma.com/community/file/1002945721703152933) legger en endringslogg synlig i Figma (enten på cover eller som egen side) – antar det gjøres manuelt. Blir kanskje litt dobbelt opp, men en copy/paste gjør kanskje nytten? Er sikkert fint å se endringer i den faktiske versjonshistorikken også.

**Reply by onordbo** [reply]
+1 på bedre beskrivelser i endringsloggen, hvis vi begynner med forslaget til @AudunBerdal som et første steg, så er det nok ganske kort vei til å visualisere dette et sted som er lett tilgjengelig for alle brukere av Jøkul?

**Comment by wkillerud** [comment]
Jeg ser vi hadde en periode med [månedlige innlegg på Jøkul-bloggen](https://jokul.fremtind.no). Med litt TLC (#2285, #2292) kan det kanskje være noe å ta opp igjen?

**Comment by onordbo** [comment]
Det finnes dokumentasjon i Figma API'et som tilsynelatende lar oss hente ut [endringer i versjonshistorikken](https://www.figma.com/developers/api#version-history) programmatisk. Kanskje vi skal se på mulighetene for å automatisk publisere disse i en relevant kanal? (Teams, Jøkulbloggen, etc) Det er nok viktig at vi tilrettelegger for målgruppen vår (designere) som ikke nødvendigvis er innom Github veldig regelmessig. Hvis vi kan løse dette parallellt med @AudunBerdal's forslag, så synes jeg vi har ett veldig godt utgangspunkt som vil kreve lite manuelt arbeid.

**Reply by idalower-zz** [reply]
Tenker at det er ulike grader av endringer. Større endringer som nye meldingsbokser og tilbakemeldingsmodul kan man gjerne flagge i flere kanaler. Også gjerne på skissestadiet, type: "vi jobber med meldingsbokser, de kommer til å se noe slik ut. Work in progress..."
Mens mindre endringer i kode eller småpirk i design kan man fortsette å snakke om i feks DS og her på Github. Er også en evig erkjennelse at vi designere ikke titter innom Github altfor ofte

**Reply by onordbo** [reply]
Det er definitivt viktig å skille mellom nivåer av endringer og utvikling i Jøkul, det er jo alltids noen ting som definitivt er viktigere å kommunisere bredt enn andre. 😄 Samtidig tror jeg også det er viktig at vi ivaretar et felles, godt sortert fora med høy skannbarhet, slik at vi gjør det enklest mulig for oss selv å diskutere behov og muligheter på tvers av teamene våre, uten å måtte gå på ekspedisjon i meldingshistorikk. Dog er jeg opptatt av at det skal være enklest mulig for alle involverte _(både de som deler, og de som søker informasjon)_. Jeg tror vi har bomma hvis vi lager en løsning som føles som enda en "rydde i boden"-oppgave man har hengende over seg.

**Comment by onordbo** [comment]
Jeg savner litt automatikk som gjør det lettere å vise historikk og aktivitet fra Figma i en Github Discussion/Issue, og på Teams _(ev. andre relevante kanaler?)_. Vi har et behov for skape en ryddigere oversikt over hva som skjer i Jøkul med jevnlige oppdateringer, så det er lettere for alle å holde seg oppdatert og bli informert. I utviklingen på kodesiden er vi gode på detaljerte oppdateringer, da Github er spesifikt bygget for dette, på designsiden finnes det ikke noen særlig god løsning som ikke er helmanuell, p.t

> Hvordan kan vi simulere en liknende opplevelse for design av Jøkul? Hvordan burde det se ut, og hva slags behov skal en slik automatikk møte?

I en perfekt verden ønsker jeg med en slags @jøkulbot-dings til både GitHub og Teams som jeg bare kan tagge i hytt og gevær som en slags designer-butler, som holder teamet oppdatert og viser frem prosessen mens den pågår.

#### Behov/ønsker/atferd/tanker/refleksjoner til hvordan design/tech-butleren vår kan oppføre seg:
- Botten burde nok ikke være helautomatisk, det skaper nok unødvendig mye kompleksitet
- Jeg vil kunne tagge (@) bot’en med en referansefil, som den da hekter seg på og sporer versjonshistorikken til
- Botten burde nok begrenses til å legge ut status-oppdateringer fra “hard saves” (commits) i Figma sin versjonshistorikk, (og kanskje eventuelle merges, review requests, nye branches, etc, avhengig av [API støtte](https://www.figma.com/developers/api))
- I en drømmeverden vil jeg få inn Figma sine Live-Embed’s rett i en Issue/Discussion, og eventuelt peke til et spesifikt sted i ett dokument, via en referanse/commit/lenke av noe slag. _(Et bilde sier mer enn tusen ord, anyone?)_ Men det ser ikke ut som det er mulig/tilgjengelig enda. Så hvor mange visuelle detaljer og differanser vi klarer å automagisk dele avhenger nok litt av Figma API’et

**Reply by wkillerud** [reply]
> I en drømmeverden vil jeg få inn Figma sine Live-Embed’s rett i en Issue/Discussion

_The future is now_* (med en [plugin til Chrome](https://github.com/camilleriluke/figma-github), kan [testes her](https://github.com/camilleriluke/figma-github/issues/1))
